ethereum_history_api/circuits/lib/src/merkle_patricia_proofs/proof_test.nr

```diff
@@ -462,4 +515,52 @@ mod verify_merkle_proof {
             cancun_transaction_proof_input.proof
         );
     }
+
+    #[test(should_fail_with="Leaf key part does not correspond to left key nibbles")]
```

Comment:
"Leaf key part does not correspond to left key nibbles"
If we want to write it like this - I would replace left with leftover as left also has left/right meaning.
But I would propose: `Key mismatch in the leaf`